src/util/options.rs

```diff
@@ -397,7 +469,21 @@ options! {
     work_perf_events:       PerfEventOptions     [env_var: true, command_line: true] [|_| cfg!(all(feature = "perf_counter", feature = "work_packet_stats"))] = PerfEventOptions {events: vec![]},
     // Measuring perf events for GC and mutators
     // TODO: Ideally this option should only be included when the features 'perf_counter' are enabled. The current macro does not allow us to do this.
-    phase_perf_events:      PerfEventOptions     [env_var: true, command_line: true] [|_| cfg!(feature = "perf_counter")] = PerfEventOptions {events: vec![]}
+    phase_perf_events:      PerfEventOptions     [env_var: true, command_line: true] [|_| cfg!(feature = "perf_counter")] = PerfEventOptions {events: vec![]},
+    // Set how to bind affinity to the GC Workers. Default thread affinity delegates to the OS scheduler. XXX: This option is currently only supported on Linux.
+    thread_affinity:        AffinityKind         [env_var: true, command_line: true] [always_valid] = AffinityKind::OsDefault,
```

Comment:
This seems cleaner. 
```rust
pub enum AffinityKind {
    OsDefault,
    Fixed(CoreId),
    #[cfg(target_os = "linux)]
    RoundRobin(Vec<CoreId>),
}
```

* `cpu_list` has to be used with `thread_affinity`. It would be better not to have a separate option for `cpu_list`. 
* Supply `CoreId` with each variant of the enum is enough.
* You can let the scheduler to create the `CpuSet` from the option. And you can put `CPU_SET_IDX` to the struct (unless you actually want to this to be global, i.e. used across multiple MMTk instances).
* `CpuSet.num_cpu` is only used to verify the option so you don't need to actually save it anywhere. You can use the validation method to do it (use your validation method instead of `always_valid`).

Reply:
Hmm fair enough. In that case would the user specify a core list as `MMTK_THREAD_AFFINITY=RoundRobin("0,3-4")`? Or I guess we could just do if it's 1 core then it's a Fixed affinity and greater than 1 core implies `RoundRobin`.

Reply:
PS Do you have any opinions on the pinning of the coordinator thread? Currently the PR will pin it, but I think it is not required (and only complicates the pinning).

Reply:
You can do something similar to the bounded/fixed nursery option you did before, something like `RoundRobin:0,3-4`.

Reply:
> Or I guess we could just do if it's 1 core then it's a Fixed affinity and greater than 1 core implies RoundRobin.

I think this is simpler (for the user). Do we envision a different method of assigning cores to threads in the future? Like random maybe? If not, then I think the above will allow for a simpler user interface.

Reply:
> PS Do you have any opinions on the pinning of the coordinator thread? Currently the PR will pin it, but I think it is not required (and only complicates the pinning).

I have no preference on this.

> Or I guess we could just do if it's 1 core then it's a Fixed affinity and greater than 1 core implies RoundRobin.

This sounds fine. Actually you don't need to deal with `Fixed` specially. `x % 1 = 0`, which is the index of the only element in the array.

Reply:
True -- we can just remove `Fixed` in favour of just `RoundRobin` then.

Reply:
@caizixian Thoughts on pinning GC coordinator thread? I'm inclined to remove it, but if you have a case for it, then I'll just leave it in.

Reply:
I've addressed your comments in https://github.com/mmtk/mmtk-core/pull/670/commits/248ba47deede2ab6603060150c147b33ebca1a90, @qinsoon. Zixian and I think that pinning GC coordinator thread is not necessary, so I've removed it.